functions.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+function toJson(array $data): ?string
+{
+    try {
+        $jsonEncoded = json_encode($data, JSON_ERROR_SYNTAX, 512);
+    } catch (JsonException $e) {
+        echo $e->getMessage();
+        return null;
+    }
+
+    return $jsonEncoded;
+}
+
+function fromJson(string $data): ?array
+{
+    try {
+        $jsonDecoded = json_decode($data, true, 512, JSON_ERROR_SYNTAX);
+    } catch (JsonException $e) {
+        echo $e->getMessage();
+        return null;
+    }
+
+    return $jsonDecoded;
+}
+
+function sendResponse(string $message, int $statusCode): void
+{
+    http_response_code($statusCode);
+
+    echo $message;
+
+    exit;
+}
+
+function processURL(string &$endpoint, array &$params, string &$method): void
+{
+    $url = parse_url($_SERVER['REQUEST_URI']);
+
+    $endpoint = str_replace('/', '', $url['path']);
+
+    $params = [];
+
+    if (isset($url['query'])) {
+        parse_str($url['query'], $params);
+    }
+
+    $method = $_SERVER['REQUEST_METHOD'];
+}
```

Comment:
this method looks incomplete.

Reply:
@AndreiOrmanji - Modified this function to not set so many variables by reference and return an associative array instead.